src/pages/journal/journal.effects.ts

```diff
@@ -68,13 +68,28 @@ export class JournalEffects {
             tap((journalData: ExaminerWorkSchedule) => this.journalProvider.saveJournalForOffline(journalData)),
             map((journalData: ExaminerWorkSchedule) => this.slotProvider.detectSlotChanges(slots, journalData)),
             map((slots: any[]) => groupBy(slots, this.slotProvider.getSlotDate)),
-            map((slots: {[k: string]: SlotItem[]}) => this.slotProvider.extendWithEmptyDays(slots)),
-            map((slots: {[k: string]: SlotItem[]}) => this.slotProvider.getRelevantSlots(slots)),
-            map((slots: {[k: string]: SlotItem[]}) =>
-              new journalActions.LoadJournalSuccess(slots,
-                                                    this.networkStateProvider.getNetworkState(),
-                                                    this.authProvider.isInUnAuthenticatedMode(),
-                                                    lastRefreshed)),
+            map((slots: { [k: string]: SlotItem[] }) => this.slotProvider.extendWithEmptyDays(slots)),
+            map((slots: { [k: string]: SlotItem[] }) => this.slotProvider.getRelevantSlots(slots)),
+            map((slots: { [k: string]: SlotItem[] }) =>
+              new journalActions.LoadJournalSuccess(
+                slots,
+                this.networkStateProvider.getNetworkState(),
+                this.authProvider.isInUnAuthenticatedMode(),
+                lastRefreshed,
+              ),
+            ),
+            catchError((err) => {
+              // For HTTP 304 NOT_MODIFIED we just use the slots we already have cached
+              if (err.status === 304) {
```

Comment:
If we aren't modifying the data do we need to run the success action as the data will just be the same right

Reply:
I tried this, but the spinner stayed up as that action is triggering the spinner dismissal.

Reply:
Just looked and it seems a pain to untangle, doing it this way won't cause any issues :)